packages/front-end/components/Holdout/HoldoutSelect.tsx

```diff
@@ -82,36 +83,42 @@ export const HoldoutSelect = ({
         setHoldout(holdoutsWithExperiment[0].id);
       }
     }
-  }, [holdoutsWithExperiment, setHoldout, loading, selectedHoldoutId]);
+  }, [
+    holdoutsWithExperiment,
+    setHoldout,
+    loading,
+    selectedHoldoutId,
+    hasHoldouts,
+  ]);
```

Comment:
Code like this always makes me nervous.  We're calling setState methods inside useEffect with a ton of dependencies.  This makes it super easy for bugs to creep in and run this code more often than we want.  This would have the effect of overwriting things in the form the user has already selected.